spec/environment_spec.rb

```diff
@@ -545,6 +550,131 @@ def promote_content(content1, env1)
     }
   end
 
+  it 'should delete consumer together with his last environment' do
+    product = create_product
+    content1 = new_content(1)
+    content2 = new_content(2)
+
+    @cp.add_content_to_product(@owner['key'], product['id'], content1['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], content2['id'])
+
+    env1 = @cp.create_environment(@owner['key'], random_string("env1"), random_string("test_env_1"))
+    env2 = @cp.create_environment(@owner['key'], random_string("env2"), random_string("test_env_2"))
+    promote_content(content1, env1)
+    promote_content(content2, env1)
+    promote_content(content2, env2)
+    consumer1 = @cp.register(random_string("consumer1"), :system, nil,
+                            facts={'system.certificate_version' => '3.3', "uname.machine" => "x86_64"}, random_string("consumer1"),
+                            @owner['key'], [], [], [{ 'id' => env1['id']}])
+    consumer2 = @cp.register(random_string("consumer2"), :system, nil,
+                            facts={'system.certificate_version' => '3.3', "uname.machine" => "x86_64"}, random_string("consumer2"),
+                            @owner['key'], [], [], [{ 'id' => env1['id']}, { 'id' => env2['id']}])
+
+    @cp.delete_environment(env1['id'])
+
+    lambda {
+      @cp.get_consumer(consumer1['uuid'])
+    }.should raise_exception(RestClient::Gone)
+
+    cons = @cp.get_consumer(consumer2['uuid'])#.should_not be nil
+    cons['environments'].length.should == 1
+  end
+
+  it 'should only enable content promoted by consumers environment' do
```

Comment:
This test is a duplicate of `'filters content not promoted to environment'` from `environment_cert_v3.rb`

Reply:
Done.